[CI] MixedClusterClientYamlTestSuiteIT fails with 'mixed os cluster found' on Windows

`MixedClusterClientYamlTestSuiteIT` tests fail with `mixed os cluster found` error, seems to be limited to Windows workers but tests failing are different each time

**Build scan**:
https://gradle-enterprise.elastic.co/s/fe7rxzbmjzvsg
**Repro line**:
`./gradlew ':qa:mixed-cluster:v7.12.0#mixedClusterTest' -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=indices.put_mapping/all_path_options/put mapping in _all index}" -Dtests.seed=2FD6F136725F1C0C -Dtests.security.manager=true -Dtests.bwc=true -Dtests.locale=fr-FR -Dtests.timezone=Australia/Canberra -Druntime.java=11`
**Reproduces locally?**:
No on mac, don't have windows to check
**Applicable branches**:
`master`
**Failure history**:
<!--
Link to build stats and possible indication of when this started failing and how often it fails
<https://build-stats.elastic.co/app/kibana>
-->
**Failure excerpt**:
```
java.lang.AssertionError: mixed os cluster found 
  |   | at __randomizedtesting.SeedInfo.seed([2FD6F136725F1C0C:A782CEECDCA371F4]:0) |  
  |   | at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.readOsFromNodesInfo(ESClientYamlSuiteTestCase.java:365) |  
  |   | at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.initAndResetContext(ESClientYamlSuiteTestCase.java:138) |  
  |   | at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) |  
  |   | at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) |  
  |   | at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) |  
  |   | at java.lang.reflect.Method.invoke(Method.java:566) |  
  |   | at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758) |  
  |   | at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:980) |  
  |   | at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996) |  
  |   | at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36) |  
  |   | at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49) |  
  |   | at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45) |  
  |   | at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48) |  
  |   | at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64) |  
  |   | at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)



```

The new debug shows this:

```
> Task :qa:mixed-cluster:v7.12.0#mixedClusterTest


org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT > test {p0=indices.create/10_basic/Create index with mappings} FAILED
    java.lang.AssertionError: mixed os cluster found: [Windows Server 2019, Windows Server 2016]
```

So it the JDK for one of the nodes in the mixed version cluster (either 7.12 or 8.0) reports that Windows Server 2019 is Windows Server 2016.  It's the problem reported in https://bugs.openjdk.java.net/browse/JDK-8211106.  That was fixed in Java 12.

The build reports runtime JDK 11:

```
=======================================
Elasticsearch Build Hamster says Hello!
  Gradle Version        : 6.8.1
  OS Info               : Windows Server 2019 10.0 (amd64)
  Runtime JDK Version   : 11 (Oracle)
  Runtime java.home     : C:\Users\jenkins\.java\openjdk11
  Gradle JDK Version    : 15 (Oracle)
  Gradle java.home      : C:\Users\jenkins\.java\openjdk15
  Random Testing Seed   : 8E287B3A40EB6619
  In FIPS 140 mode      : false
=======================================
```

From the server side logs, this is what is happening:

```
bash-3.2$ grep -C 1 'using bundled JDK'  */logs/v7.12.0.log
v7.12.0-0/logs/v7.12.0.log-[2021-02-03T20:06:20,891][INFO ][o.e.n.Node               ] [v7.12.0-0] version[7.12.0-SNAPSHOT], pid[10452], build[oss/zip/957b9c5c5c96b7b58f377c6389be7cd23a4a3289/2021-02-03T19:52:40.755655Z], OS[Windows Server 2019/10.0/amd64], JVM[AdoptOpenJDK/OpenJDK 64-Bit Server VM/15.0.1/15.0.1+9]
v7.12.0-0/logs/v7.12.0.log:[2021-02-03T20:06:20,895][INFO ][o.e.n.Node               ] [v7.12.0-0] JVM home [C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2019\qa\mixed-cluster\build\testclusters\v7.12.0-0\distro\7.12.0-OSS\jdk], using bundled JDK [true]
v7.12.0-0/logs/v7.12.0.log-[2021-02-03T20:06:20,897][INFO ][o.e.n.Node               ] [v7.12.0-0] JVM arguments [-Des.networkaddress.cache.ttl=60, -Des.networkaddress.cache.negative.ttl=10, -XX:+AlwaysPreTouch, -Xss1m, -Djava.awt.headless=true, -Dfile.encoding=UTF-8, -Djna.nosys=true, -XX:-OmitStackTraceInFastThrow, -XX:+ShowCodeDetailsInExceptionMessages, -Dio.netty.noUnsafe=true, -Dio.netty.noKeySetOptimization=true, -Dio.netty.recycler.maxCapacityPerThread=0, -Dio.netty.allocator.numDirectArenas=0, -Dlog4j.shutdownHookEnabled=false, -Dlog4j2.disable.jmx=true, -Djava.locale.providers=SPI,COMPAT, -XX:+UseG1GC, -Djava.io.tmpdir=C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2019\qa\mixed-cluster\build\testclusters\v7.12.0-0\tmp, -XX:+HeapDumpOnOutOfMemoryError, -XX:HeapDumpPath=logs, -XX:ErrorFile=logs\hs_err_pid%p.log, -Xlog:gc*,gc+age=trace,safepoint:file=logs\gc.log:utctime,pid,tags:filecount=32,filesize=64m, -Xms512m, -Xmx512m, -ea, -esa, -Dio.netty.leakDetection.level=paranoid, -XX:MaxDirectMemorySize=268435456, -XX:G1HeapRegionSize=4m, -XX:InitiatingHeapOccupancyPercent=30, -XX:G1ReservePercent=15, -Delasticsearch, -Des.path.home=C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2019\qa\mixed-cluster\build\testclusters\v7.12.0-0\distro\7.12.0-OSS, -Des.path.conf=C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2019\qa\mixed-cluster\build\testclusters\v7.12.0-0\config, -Des.distribution.flavor=oss, -Des.distribution.type=zip, -Des.bundled_jdk=true]
--
--
v7.12.0-0/logs/v7.12.0.log-[2021-02-03T20:06:41,249][INFO ][o.e.n.Node               ] [v7.12.0-0] version[8.0.0-SNAPSHOT], pid[7212], build[oss/zip/9e052311b9fd2397cd80d990b48a0042ec7531b9/2021-02-03T17:51:11.468126900Z], OS[Windows Server 2016/10.0/amd64], JVM[Oracle Corporation/OpenJDK 64-Bit Server VM/11.0.2/11.0.2+7]
v7.12.0-0/logs/v7.12.0.log:[2021-02-03T20:06:41,257][INFO ][o.e.n.Node               ] [v7.12.0-0] JVM home [C:\Users\jenkins\.java\openjdk11], using bundled JDK [false]
v7.12.0-0/logs/v7.12.0.log-[2021-02-03T20:06:41,258][INFO ][o.e.n.Node               ] [v7.12.0-0] JVM arguments [-Des.networkaddress.cache.ttl=60, -Des.networkaddress.cache.negative.ttl=10, -XX:+AlwaysPreTouch, -Xss1m, -Djava.awt.headless=true, -Dfile.encoding=UTF-8, -Djna.nosys=true, -XX:-OmitStackTraceInFastThrow, -Dio.netty.noUnsafe=true, -Dio.netty.noKeySetOptimization=true, -Dio.netty.recycler.maxCapacityPerThread=0, -Dio.netty.allocator.numDirectArenas=0, -Dlog4j.shutdownHookEnabled=false, -Dlog4j2.disable.jmx=true, -Djava.locale.providers=SPI,COMPAT, -XX:+UseConcMarkSweepGC, -XX:CMSInitiatingOccupancyFraction=75, -XX:+UseCMSInitiatingOccupancyOnly, -Djava.io.tmpdir=C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2019\qa\mixed-cluster\build\testclusters\v7.12.0-0\tmp, -XX:+HeapDumpOnOutOfMemoryError, -XX:HeapDumpPath=logs, -XX:ErrorFile=logs\hs_err_pid%p.log, -Xlog:gc*,gc+age=trace,safepoint:file=logs\gc.log:utctime,pid,tags:filecount=32,filesize=64m, -Xms512m, -Xmx512m, -ea, -esa, -Dio.netty.leakDetection.level=paranoid, -XX:MaxDirectMemorySize=268435456, -Delasticsearch, -Des.path.home=C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2019\qa\mixed-cluster\build\testclusters\v7.12.0-0\distro\8.0.0-OSS, -Des.path.conf=C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2019\qa\mixed-cluster\build\testclusters\v7.12.0-0\config, -Des.distribution.flavor=oss, -Des.distribution.type=zip, -Des.bundled_jdk=true]
--
--
v7.12.0-1/logs/v7.12.0.log-[2021-02-03T20:06:22,753][INFO ][o.e.n.Node               ] [v7.12.0-1] version[7.12.0-SNAPSHOT], pid[3392], build[oss/zip/957b9c5c5c96b7b58f377c6389be7cd23a4a3289/2021-02-03T19:52:40.755655Z], OS[Windows Server 2019/10.0/amd64], JVM[AdoptOpenJDK/OpenJDK 64-Bit Server VM/15.0.1/15.0.1+9]
v7.12.0-1/logs/v7.12.0.log:[2021-02-03T20:06:22,759][INFO ][o.e.n.Node               ] [v7.12.0-1] JVM home [C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2019\qa\mixed-cluster\build\testclusters\v7.12.0-1\distro\7.12.0-OSS\jdk], using bundled JDK [true]
v7.12.0-1/logs/v7.12.0.log-[2021-02-03T20:06:22,761][INFO ][o.e.n.Node               ] [v7.12.0-1] JVM arguments [-Des.networkaddress.cache.ttl=60, -Des.networkaddress.cache.negative.ttl=10, -XX:+AlwaysPreTouch, -Xss1m, -Djava.awt.headless=true, -Dfile.encoding=UTF-8, -Djna.nosys=true, -XX:-OmitStackTraceInFastThrow, -XX:+ShowCodeDetailsInExceptionMessages, -Dio.netty.noUnsafe=true, -Dio.netty.noKeySetOptimization=true, -Dio.netty.recycler.maxCapacityPerThread=0, -Dio.netty.allocator.numDirectArenas=0, -Dlog4j.shutdownHookEnabled=false, -Dlog4j2.disable.jmx=true, -Djava.locale.providers=SPI,COMPAT, -XX:+UseG1GC, -Djava.io.tmpdir=C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2019\qa\mixed-cluster\build\testclusters\v7.12.0-1\tmp, -XX:+HeapDumpOnOutOfMemoryError, -XX:HeapDumpPath=logs, -XX:ErrorFile=logs\hs_err_pid%p.log, -Xlog:gc*,gc+age=trace,safepoint:file=logs\gc.log:utctime,pid,tags:filecount=32,filesize=64m, -Xms512m, -Xmx512m, -ea, -esa, -Dio.netty.leakDetection.level=paranoid, -XX:MaxDirectMemorySize=268435456, -XX:G1HeapRegionSize=4m, -XX:InitiatingHeapOccupancyPercent=30, -XX:G1ReservePercent=15, -Delasticsearch, -Des.path.home=C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2019\qa\mixed-cluster\build\testclusters\v7.12.0-1\distro\7.12.0-OSS, -Des.path.conf=C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2019\qa\mixed-cluster\build\testclusters\v7.12.0-1\config, -Des.distribution.flavor=oss, -Des.distribution.type=zip, -Des.bundled_jdk=true]
--
--
v7.12.0-1/logs/v7.12.0.log-[2021-02-03T20:06:55,877][INFO ][o.e.n.Node               ] [v7.12.0-1] version[8.0.0-SNAPSHOT], pid[12152], build[oss/zip/9e052311b9fd2397cd80d990b48a0042ec7531b9/2021-02-03T17:51:11.468126900Z], OS[Windows Server 2016/10.0/amd64], JVM[Oracle Corporation/OpenJDK 64-Bit Server VM/11.0.2/11.0.2+7]
v7.12.0-1/logs/v7.12.0.log:[2021-02-03T20:06:55,885][INFO ][o.e.n.Node               ] [v7.12.0-1] JVM home [C:\Users\jenkins\.java\openjdk11], using bundled JDK [false]
v7.12.0-1/logs/v7.12.0.log-[2021-02-03T20:06:55,885][INFO ][o.e.n.Node               ] [v7.12.0-1] JVM arguments [-Des.networkaddress.cache.ttl=60, -Des.networkaddress.cache.negative.ttl=10, -XX:+AlwaysPreTouch, -Xss1m, -Djava.awt.headless=true, -Dfile.encoding=UTF-8, -Djna.nosys=true, -XX:-OmitStackTraceInFastThrow, -Dio.netty.noUnsafe=true, -Dio.netty.noKeySetOptimization=true, -Dio.netty.recycler.maxCapacityPerThread=0, -Dio.netty.allocator.numDirectArenas=0, -Dlog4j.shutdownHookEnabled=false, -Dlog4j2.disable.jmx=true, -Djava.locale.providers=SPI,COMPAT, -XX:+UseConcMarkSweepGC, -XX:CMSInitiatingOccupancyFraction=75, -XX:+UseCMSInitiatingOccupancyOnly, -Djava.io.tmpdir=C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2019\qa\mixed-cluster\build\testclusters\v7.12.0-1\tmp, -XX:+HeapDumpOnOutOfMemoryError, -XX:HeapDumpPath=logs, -XX:ErrorFile=logs\hs_err_pid%p.log, -Xlog:gc*,gc+age=trace,safepoint:file=logs\gc.log:utctime,pid,tags:filecount=32,filesize=64m, -Xms512m, -Xmx512m, -ea, -esa, -Dio.netty.leakDetection.level=paranoid, -XX:MaxDirectMemorySize=268435456, -Delasticsearch, -Des.path.home=C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2019\qa\mixed-cluster\build\testclusters\v7.12.0-1\distro\8.0.0-OSS, -Des.path.conf=C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2019\qa\mixed-cluster\build\testclusters\v7.12.0-1\config, -Des.distribution.flavor=oss, -Des.distribution.type=zip, -Des.bundled_jdk=true]
--
--
v7.12.0-2/logs/v7.12.0.log-[2021-02-03T20:06:24,714][INFO ][o.e.n.Node               ] [v7.12.0-2] version[7.12.0-SNAPSHOT], pid[6340], build[oss/zip/957b9c5c5c96b7b58f377c6389be7cd23a4a3289/2021-02-03T19:52:40.755655Z], OS[Windows Server 2019/10.0/amd64], JVM[AdoptOpenJDK/OpenJDK 64-Bit Server VM/15.0.1/15.0.1+9]
v7.12.0-2/logs/v7.12.0.log:[2021-02-03T20:06:24,721][INFO ][o.e.n.Node               ] [v7.12.0-2] JVM home [C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2019\qa\mixed-cluster\build\testclusters\v7.12.0-2\distro\7.12.0-OSS\jdk], using bundled JDK [true]
v7.12.0-2/logs/v7.12.0.log-[2021-02-03T20:06:24,723][INFO ][o.e.n.Node               ] [v7.12.0-2] JVM arguments [-Des.networkaddress.cache.ttl=60, -Des.networkaddress.cache.negative.ttl=10, -XX:+AlwaysPreTouch, -Xss1m, -Djava.awt.headless=true, -Dfile.encoding=UTF-8, -Djna.nosys=true, -XX:-OmitStackTraceInFastThrow, -XX:+ShowCodeDetailsInExceptionMessages, -Dio.netty.noUnsafe=true, -Dio.netty.noKeySetOptimization=true, -Dio.netty.recycler.maxCapacityPerThread=0, -Dio.netty.allocator.numDirectArenas=0, -Dlog4j.shutdownHookEnabled=false, -Dlog4j2.disable.jmx=true, -Djava.locale.providers=SPI,COMPAT, -XX:+UseG1GC, -Djava.io.tmpdir=C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2019\qa\mixed-cluster\build\testclusters\v7.12.0-2\tmp, -XX:+HeapDumpOnOutOfMemoryError, -XX:HeapDumpPath=logs, -XX:ErrorFile=logs\hs_err_pid%p.log, -Xlog:gc*,gc+age=trace,safepoint:file=logs\gc.log:utctime,pid,tags:filecount=32,filesize=64m, -Xms512m, -Xmx512m, -ea, -esa, -Dio.netty.leakDetection.level=paranoid, -XX:MaxDirectMemorySize=268435456, -XX:G1HeapRegionSize=4m, -XX:InitiatingHeapOccupancyPercent=30, -XX:G1ReservePercent=15, -Delasticsearch, -Des.path.home=C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2019\qa\mixed-cluster\build\testclusters\v7.12.0-2\distro\7.12.0-OSS, -Des.path.conf=C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2019\qa\mixed-cluster\build\testclusters\v7.12.0-2\config, -Des.distribution.flavor=oss, -Des.distribution.type=zip, -Des.bundled_jdk=true]
--
--
v7.12.0-3/logs/v7.12.0.log-[2021-02-03T20:06:26,801][INFO ][o.e.n.Node               ] [v7.12.0-3] version[7.12.0-SNAPSHOT], pid[11012], build[oss/zip/957b9c5c5c96b7b58f377c6389be7cd23a4a3289/2021-02-03T19:52:40.755655Z], OS[Windows Server 2019/10.0/amd64], JVM[AdoptOpenJDK/OpenJDK 64-Bit Server VM/15.0.1/15.0.1+9]
v7.12.0-3/logs/v7.12.0.log:[2021-02-03T20:06:26,807][INFO ][o.e.n.Node               ] [v7.12.0-3] JVM home [C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2019\qa\mixed-cluster\build\testclusters\v7.12.0-3\distro\7.12.0-OSS\jdk], using bundled JDK [true]
v7.12.0-3/logs/v7.12.0.log-[2021-02-03T20:06:26,810][INFO ][o.e.n.Node               ] [v7.12.0-3] JVM arguments [-Des.networkaddress.cache.ttl=60, -Des.networkaddress.cache.negative.ttl=10, -XX:+AlwaysPreTouch, -Xss1m, -Djava.awt.headless=true, -Dfile.encoding=UTF-8, -Djna.nosys=true, -XX:-OmitStackTraceInFastThrow, -XX:+ShowCodeDetailsInExceptionMessages, -Dio.netty.noUnsafe=true, -Dio.netty.noKeySetOptimization=true, -Dio.netty.recycler.maxCapacityPerThread=0, -Dio.netty.allocator.numDirectArenas=0, -Dlog4j.shutdownHookEnabled=false, -Dlog4j2.disable.jmx=true, -Djava.locale.providers=SPI,COMPAT, -XX:+UseG1GC, -Djava.io.tmpdir=C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2019\qa\mixed-cluster\build\testclusters\v7.12.0-3\tmp, -XX:+HeapDumpOnOutOfMemoryError, -XX:HeapDumpPath=logs, -XX:ErrorFile=logs\hs_err_pid%p.log, -Xlog:gc*,gc+age=trace,safepoint:file=logs\gc.log:utctime,pid,tags:filecount=32,filesize=64m, -Xms512m, -Xmx512m, -ea, -esa, -Dio.netty.leakDetection.level=paranoid, -XX:MaxDirectMemorySize=268435456, -XX:G1HeapRegionSize=4m, -XX:InitiatingHeapOccupancyPercent=30, -XX:G1ReservePercent=15, -Delasticsearch, -Des.path.home=C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2019\qa\mixed-cluster\build\testclusters\v7.12.0-3\distro\7.12.0-OSS, -Des.path.conf=C:\Users\jenkins\workspace\elastic+elasticsearch+master+multijob-windows-compatibility\os\windows-2019\qa\mixed-cluster\build\testclusters\v7.12.0-3\config, -Des.distribution.flavor=oss, -Des.distribution.type=zip, -Des.bundled_jdk=true]
```

So the nodes in the cluster start off on 7.12 using their bundled JDK, which is Java 12.  Then they get upgraded to 8.0 and use the runtime JDK reported by the build, which is Java 11.  So in the mixed version state the Java 11 8.0 nodes are reporting Windows Server 2016 and the Java 12 7.12 nodes are reporting Windows Server 2019.

In reality these tests should can never run on multiple operating systems simultaneously, as they are running on a single machine.  Presumably there are limited cases where multiple operating systems will be reported, likely limited to Windows's inaccurate version reporting through the Windows API.  In this case sorting alphabetically and taking the last item is likely to return the most accurate version, so I will change the code to do that.

--- a/test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ESClientYamlSuiteTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ESClientYamlSuiteTestCase.java
@@ -51,6 +51,8 @@ import java.util.Locale;
 import java.util.Map;
 import java.util.Map.Entry;
 import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
 
 /**
  * Runs a suite of yaml tests shared with all the official Elasticsearch
@@ -339,7 +341,7 @@ public abstract class ESClientYamlSuiteTestCase extends ESRestTestCase {
         final Request request = new Request("GET", "/_nodes/os");
         Response response = restClient.performRequest(request);
         ClientYamlTestResponse restTestResponse = new ClientYamlTestResponse(response);
-        Set<String> osPrettyNames = new HashSet<>();
+        SortedSet<String> osPrettyNames = new TreeSet<>();
 
         @SuppressWarnings("unchecked")
         final Map<String, Object> nodes = (Map<String, Object>) restTestResponse.evaluate("nodes");
@@ -351,8 +353,16 @@ public abstract class ESClientYamlSuiteTestCase extends ESRestTestCase {
             osPrettyNames.add((String) XContentMapValues.extractValue("os.pretty_name", nodeInfo));
         }
 
-        assert osPrettyNames.size() == 1 : "mixed os cluster found: " + osPrettyNames;
-        return osPrettyNames.iterator().next();
+        assert osPrettyNames.isEmpty() == false : "no os found";
+
+        // Although in theory there should only be one element as all nodes are running on the same machine,
+        // in reality there can be two in mixed version clusters if different Java versions report the OS
+        // name differently. This has been observed to happen on Windows, where Java needs to be updated to
+        // recognize new Windows versions, and until this update has been done the newest version of Windows
+        // is reported as the previous one. In this case taking the last alphabetically is likely to be most
+        // accurate, for example if "Windows Server 2016" and "Windows Server 2019" are reported by different
+        // Java versions then Windows Server 2019 is likely to be correct.
+        return osPrettyNames.last();
     }
 
     protected RequestOptions getCatNodesVersionMasterRequestOptions() {

